doc/src/manual/performance-tips.md

```diff
@@ -149,7 +151,38 @@ its algorithmic aspects (see [Pre-allocating outputs](@ref)).
     For more serious benchmarking, consider the [BenchmarkTools.jl](https://github.com/JuliaCI/BenchmarkTools.jl)
     package which among other things evaluates the function multiple times in order to reduce noise.
 
-## [Tools](@id tools)
+### Break functions into multiple definitions
```

Comment:
honestly this section should probably be removed (or substantially reworked). I agree that the 2nd version is better, but the performance of them will be identical since Julia will use the if statements to get type stability and inline anyway.

Reply:
I agree. This seems more like a "design tip" than a "performance tip". So perhaps it should be rehomed to the Style Guide. Maybe around https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-strange-type-Unions?

Reply:
I voluntarily restricted myself to moving stuff around for this PR in order to facilitate review. If it seems like a unanimous improvement maybe we could merge and discuss the actual content later?